webroot/adm/enclave-gcp-v2.html

```diff
@@ -0,0 +1,191 @@
+  <option value="true">True</option>
+</select>
+<br>
+<br>
+
+<h3>Operations</h3>
+
+<ul>
+  <li><a href="#" id="doGenerate">Generate Enclave ID and gcloud command</a></li>
+</ul>
+<br>
+
+<h3>Outputs</h3>
+
+<label for="gcloudCli">Example gcloud command to bring up Confidential Space VM:</label>
+<br>
+<textarea id="gcloudCli" name="gcloudCli" rows="10" cols="120" readonly>
+</textarea>
+<br>
+<br>
+
+<label for="enclaveId">Generated GCP Enclave ID:</label>
+<br>
+<pre><label id="enclaveId">TBD</label></pre>
+<br>
+<br>
+
+<script language="JavaScript">
+  const gcloudCli = '$ gcloud compute instances create <INSTANCE_NAME> \\\n\
+    --confidential-compute \\\n\
+    --shielded-secure-boot \\\n\
+    --maintenance-policy Terminate \\\n\
+    --scopes cloud-platform \\\n\
+    --image-project confidential-space-images \\\n\
+    --image-family <IMAGE_FAMILY> \\\n\
+    --service-account <SERVICE_ACCOUNT> \\\n\
+    --metadata ^~^tee-image-reference=ghcr.io/iabtechlab/uid2-operator@<IMAGE_SHA>~tee-restart-policy=Never~tee-env-API_TOKEN=<API_TOKEN>~tee-env-DEPLOYMENT_ENVIRONMENT=<DEPLOYMENT_ENVIRONMENT>~tee-env-IDENTITY_SCOPE=<IDENTITY_SCOPE>'
+
+  $('#gcloudCli').val(gcloudCli);
+
+  $(document).ready(function () {
+    onEnvChange();
+    $('#envId').on('change', function () {
+      onEnvChange();
+    });
+
+    $('#doGenerate').on('click', function () {
+      onGenerate();
+    });
+  });
+
+  function onEnvChange() {
+    console.log('onEnvChange');
+    const env = $('#envId').val();
+    const $debugMode = $('#debugMode');
+    const $overrides = $('#optionalOverrides');
+    console.log(env);
+    if (env === 'prod') {
+      $debugMode.prop('disabled', true);
+      $debugMode.val('false');
+      $overrides.hide();
+    } else {
+      $debugMode.prop('disabled', false);
+      $debugMode.val('false');
+      $overrides.show();
+    }
+  }
+
+  function onGenerate() {
+    const apiToken = $('#apiToken').val();
+    if (!apiToken) {
+      alert("Operator Api Key is empty");
+      return;
+    }
+    let imageSha = $('#imageId').val();
+    if (!imageSha) {
+      alert("Operator Docker Image ID is empty");
+      return;
+    }
+    if (!imageSha.startsWith('sha256:')) {
+      imageSha = 'sha256:' + imageSha;
+    }
+
+    const env = $('#envId').val();
+    const identityScope = $('#identityScope').val();
+    const isDebug = $('#debugMode').val();
+
+    // generate and show enclaveId
+    generateEnclaveId(imageSha, isDebug).then(enclaveId => {
+      console.log("isDebug:" + isDebug + ", imageSha:" + imageSha);
+      console.log("enclaveId:" + enclaveId);
+      $('#enclaveId').text(enclaveId);
+    });
+
+    // update gcloud Cli
+    const imageFamily = isDebug ? 'confidential-space-debug' : 'confidential-space';
+    const cli = gcloudCli
+            .replaceAll('<IMAGE_FAMILY>', imageFamily)
+            .replaceAll('<IMAGE_SHA>', imageSha)
+            .replaceAll('<API_TOKEN>', apiToken)
```

Comment:
The API token is considered sensitive and we never send it via email, or in any text files. So the resulting gcloud CLI command must still have <API_TOKEN> (the actual string, not a replaced value). As the API key isn't used for the generation of the enclave Id, there is no reason for it to be collected on this page.

Reply:
ACK, will update.